Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/DeleteCaseTask.java

```diff
@@ -0,0 +1,850 @@
+                                /*
+                                 * Delete the input directory if it is empty.
+                                 */
+                                final Path inputDirectoryPath = manifestFilePath.getParent();
+                                final File inputDirectory = inputDirectoryPath.toFile();
+                                File[] files = inputDirectory.listFiles();
+                                logger.log(Level.INFO, String.format("Deleting empty input directory %s for %s", inputDirectoryPath, caseNodeData.getDisplayName()));
+                                if (files == null || files.length == 0) {
+                                    if (!inputDirectory.delete()) {
+                                        logger.log(Level.WARNING, String.format("Failed to delete empty input directory %s for %s", inputDirectoryPath, caseNodeData.getDisplayName()));
+                                    }
+                                }
+
+                            } else {
+                                logger.log(Level.WARNING, String.format("Failed to delete manifest file %s for %s", manifestFilePath, caseNodeData.getDisplayName()));
+                                allInputDeleted = false;
+                            }
+                        }
+                    } else {
+                        logger.log(Level.WARNING, String.format("Failed to parse manifest file %s for %s", manifestFilePath, caseNodeData.getDisplayName()));
+                    }
+                } else {
+                    logger.log(Level.WARNING, String.format("Did not find manifest file %s for %s", manifestFilePath, caseNodeData.getDisplayName()));
+                }
+            }
+            if (allInputDeleted) {
+                setDeletedItemFlag(CaseNodeData.DeletedFlags.DATA_SOURCES);
+            }
+
+        } catch (TskCoreException | UserPreferencesException ex) {
+            logger.log(Level.INFO, String.format("Failed to open the case database for %s", caseNodeData.getDisplayName()), ex);
+
+        } finally {
+            if (caseDb != null) {
+                caseDb.close();
+            }
+        }
+    }
+
+    /**
+     * Locates and deletes the data source files referenced by a manifest.
+     *
+     * @param manifest    A manifest.
+     * @param dataSources The data sources in the case as obtained from the case
+     *                    database.
+     *
+     * @return True if all of the data source files werre deleted, false
+     *         otherwise.
+     */
+    @NbBundle.Messages({
+        "# {0} - data source name", "# {1} - device id", "DeleteCaseTask.progress.deletingDataSource=Deleting data source {0} with device id {1}...",})
+    private boolean deleteDataSources(Manifest manifest, List<DataSource> dataSources) {
+        final String dataSourceFileName = manifest.getDataSourceFileName();
+        final String dataSourceDeviceId = manifest.getDeviceId();
```

Comment:
Isn't the device ID optional in the manifest file?

Reply:
It looks like while DeviceID is optional for the manifest file, it is required to construct the Manifest object so we I think its probably generated after reading the file before constructing the object

Reply:
Yes just confirmed that if there is no devId field in the manifest it will auto-generate one.

Reply:
Continuing discussion in JIRA-4838